treeherder/model/management/commands/cycle_data.py

```diff
@@ -126,7 +142,12 @@ def cycle(self):
                     repository_id=signature.repository_id,  # leverages (repository, signature) compound index
                     signature_id=signature.id,
                 ).exists():
+                    self.deleted_signatures.append(signature)
```

Comment:
You're saving this signature for later referencing, when sending the actual email. But you're deleting the signature before the email is sent. So there's nothing left to reference.